### DISCUSSION TITLE
Release of a stable version

### DISCUSSION THREAD

**Discussion by kcofoni** [kcofoni/ha-netro-watering#6] [Announcements]
### Hello all, 

Thank you to all users for cloning this repository 👍 

### Issues ? new features ?
I hope you enjoy this integration with your _Netro system_ and you don't encounter any particular difficulty. 
Do not hesitate [to post an issue](https://github.com/kcofoni/ha-netro-watering/issues) if necessary or to discuss about new features that you would like to see implemented in future versions.

### A stable version to come...
**All the functions that I wanted to develop when I decided to make this integration are now in the box** with [version 0.9.6](https://github.com/kcofoni/ha-netro-watering/releases/tag/0.9.6). I try to take particular care in the tests carried out before publishing, but I am not immune to some setbacks. Also and, insofar as I would wish **obtaining a proven reliability before creating a first stable version**, I thank you for your possible remarks in this perspective. I intend to let a week pass before releasing the _stable version_, unless your possible issues lead me to time-consuming changes.

### And after...
In my opinion, there are still a few improvements to provide around this integration. In particular, on the frontend part, I would like a suitable card, specific to this integration, to be produced. I admit that I am not a super specialist in the matter so if a good soul among you had something to offer, we would all be happy to benefit from it. As I said above I don't plan any improvements on the backend part, except perhaps creating a _weather report_ custom service which would be a direct wrapping of the [API](http://www.netrohome.com/en/shop/articles/10) on this point and which would allow you to configure Netro with your favorite weather forecast provider outside of Netro's list.

### Your feedback !
In any case I would be happy to have a feedback from you. Also do not hesitate to leave a message whether positive or negative, the main thing is to have your opinion and to move forward together.

Thanks again ! :-)

kcofoni

**Comment by kenwiens** [comment]
HI - first off - I am so pleased to see this integration.  I was considering moving to a different sprinkler system as Netro hasn't shown any interest in providing an integration.  Thanks for doing all this work!

My question - normally integrations show up in HAC's - but I don't see this one (under "Explore & Download Repositories").  You also mention adding it via the "settings tab" in HACS - and I don't have such a tab. (running HA 2023-5.4 and HACS 1.32.1).  I plan to follow the manual approach - but thought I should mention this (maybe it's me - but if not it could be confusing to others).

**Reply by kenwiens** [reply]
I have now tried the manual configuration without success.  Under the /config directory I created the directory ha-netro-watering-main, and within that are the files and folders unzipped from the archive.

I rebooted and attempted to add the integration - but there was nothing it could find - either as an existing integration or via "Add integration" .   I feel like I am missing some steps?

**Reply by kenwiens** [reply]
I received some advice:  1) go into "custom components" under HACS integrations and enter the URL provided, and then select "integrations".  After rebooting you can "download the integration.  It will not put the code into /config but rather config/custom_comnponents.  It will also rename the directory where it puts netro to "netro_watering" (from ha-netro-watering-main)

After this, you can go to integrations and run the integration which generates an error:  "Config flow could not be loaded: 500 Internal Server Error Server got itself in trouble".  I can't get beyond this point.  Reboot and try again and same results.

**Reply by kcofoni** [reply]
Hi kenwiens,

Thank you for your post. I am always happy to be in contact with Netro users.

The integration is very recent and it takes several weeks to appear natively in the integrations offered by HACS. I made the request and I imagine that my PR will soon be successful. In meantime the best way is to add custom repository "https://github.com/kcofoni/ha-netro-watering.git" as "Integration" in the settings tab of HACS. This is probably more or less what you have done at the end (your last post above).

As soon as you will have applied one of the fix I recommended for your issue #12, you should be able to take advantage of the integration. 
Enjoy !

**Reply by kenwiens** [reply]
Moving to 2023.6.2 resolved the issues and I am now up and running.  Looks like a great integration!

Over the years I have purchased many of the Netro Whisperer moisture sensors - and unfortunately I have had very poor success with them.  Even Netro tech support has given up trying to make them work for me (and I think my Netro implementation is pretty basic) - so - I really want to take advantage of the "set moisture" service in your integration.

I have recently acquired and integrated the zigbee soil moisture and temperature meter that reports both these metrics regularly into HA.

https://www.amazon.com/dp/B0BWMXQH8L?psc=1&ref=ppx_yo2ov_dt_b_product_details

Any thoughts on how I should best get the data from this sensor into Netro? I configured the set moisture service to run from a button and it looks from the code like I should be able to get the value from a template instead of manually providing it.  Is this correct?  Any thoughts before I proceed down this path?

Is the "humidity" that you provide the same as "soil moisture" - I suspect not as the values in this field differ from the soil moisture in the Netro app.  It doesn't look like I can see the soil moisture level in HA.  Is this correct?   

Perhaps a future enhancement to the integration would allow a user to easily add sensors

Anyway - love the work you have done and can't wait to take more advantage of it. If you get a chance to comment on my above thoughts / questions - it will be greatly appreciated! 

![image](https://github.com/kcofoni/ha-netro-watering/assets/35972142/1f7937dd-2d55-4c01-94fa-cd855e0fbdd7)

**Reply by kcofoni** [reply]
I'm glad you managed to get the integration working.

Many things in your message:

1. how to get the soil humidity level given by your zigbee sensor and provide it to Netro
As you guessed, it is by using the "set moisture" service that you will be able to very easily carry out the operation.
The state of the entity representing _humidity_ will simply be mentioned as an input parameter when invoking the service. And all this will end up in an `automation ` (`settings->automations & scenes->automation`) like below:

```
alias: Set moisture example
description: this set the moisture % to Netro from my zigbee sensor every two hours
trigger:
  - platform: time_pattern
    hours: /2
action:
  - service: netro_watering.set_moisture
    data:
      zone_id: 80ebf5baa5d589e9fad20e964c8da7ad
      moisture: "{{ states(soilmoisture_zigbee_entity) }}"
variables:
  soilmoisture_zigbee_entity: sensor.mysensor_humidity
mode: single
```
Of course you have to provide the actual name of your zigbee sensor humidity entity for `moisture ` parameter instead of `sensor.mysensor_humidity`.

The `zone_id parameter `points to the device representing the zone to which you want to assign the humidity value.
To obtain this id, the easiest way is to use the UI when creating your automation as illustrated by the screen shot below:

![image](https://github.com/kcofoni/ha-netro-watering/assets/75487677/2d0626a7-6383-4fd6-ba59-58c2aa2b0261)

As soon as you completed your new automation, run it and check the traces provided by the UI in the automations screen and also you can check the log file (home-assistant.log). 

2. Regarding humidity we are talking about in the set moisture service, I let you appreciate the [description given by Netro](https://www.netrohome.com/en/shop/user_guides/7): 

>  POST /npa/v1/set_moisture.json
> 
> This API is for setting zone moisture, overriding system estimated moisture.
> Params
> 
>     zones: (optional), specify the zones to water, array of zone index. Water all zones if not specified.
>     moisture: moisture level, in the range of [0,100]
> 

I hope it helps :smile:

**Reply by kenwiens** [reply]
Thanks - I'm away but hope to play more with this soon!

**Reply by kenwiens** [reply]
I did this today and it appears to work.  Note that I have 2 watering zones that overlap so I use the same sensor for both.  I have a question - I am not aware of what the "variables" construct in the automation is doing.  I can't seem to find a good description online either.  Can you explain it to me please?  I mainly use helpers in my automations..  Here is my automation:

alias: Netro Soil Moisture Back Yard
description: load netro soil moisture measurement from zigbee backyard sensor
trigger:
  - platform: time_pattern
    hours: /2
condition: []
action:
  - service: netro_watering.set_moisture
    data:
      zone_id: a606eb663fee422bf490c1d749635c78
      moisture: "{{ states('sensor.soil_moisture_meter_soil_moisture') }}"
  - service: netro_watering.set_moisture
    data:
      zone_id: 6db470a88ac5b0851fa8a91e37f6495e
      moisture: "{{ states('sensor.soil_moisture_meter_soil_moisture') }}"
mode: single
variables:
  sensor.soil_moisture_meter_soil_moisture: sensor.mysensor_humidity

**Reply by kcofoni** [reply]
The variable is just a placeholder for hosting the name of your entity (i.e. 'sensor.soil_moisture_meter_soil_moisture'). 
Since you did not mention the variable in the invocation of the states function (like I did above) but the entity directly it has no effect on your automation. So, you may delete the variables section i.e. the two lines below:
```
variables:
sensor.soil_moisture_meter_soil_moisture: sensor.mysensor_humidity
```

**Comment by kenwiens** [comment]
I have 4 zones.  For a dashboard, I'd like to show the earliest of all 4 "next watering start times".  Can you suggest a clean way of doing this? (I'd then extend this to items like - the most recent of the  last watering times, etc)

thanks,
Ken

**Reply by kcofoni** [reply]
Is it something like this you would like to get ?

![watering schedules](https://github.com/kcofoni/ha-netro-watering/assets/75487677/d83edd99-8daf-4549-9370-4e0e7c466b2b)

I have three zones and the card above just give me the last run (upper part) and the next coming run for each zone (lower part).
